File: Lezioni/10.La-gestione-dei-menu-Overflow/AppGestioneMenuOverflow/app/src/main/java/com/example/myoverflowmenuapp/MainActivity.java
package com.example.myoverflowmenuapp;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Collegamento della Toolbar come ActionBar
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflating del menu XML
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Gestione click sugli item
        int itemId = item.getItemId();
        if (itemId == R.id.action_search) {
            Toast.makeText(this, "Hai cliccato Cerca", Toast.LENGTH_SHORT).show();
            return true;
        } else if (itemId == R.id.action_settings) {
            Toast.makeText(this, "Hai cliccato Impostazioni", Toast.LENGTH_SHORT).show();
            return true;
        } else if (itemId == R.id.action_about) {
            Toast.makeText(this, "Hai cliccato Info", Toast.LENGTH_SHORT).show();
            return true;
        } else {
            return super.onOptionsItemSelected(item);
        }
    }
}

File: Lezioni/01.Realizzazione-e-test-di-una-semplice-app/Esercizi/2.Vincoli_di_stato/app/src/main/java/com/example/a2vincoli_di_stato/MainActivity.kt
package com.example.a2vincoli_di_stato

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.Button
import android.widget.TextView
import android.widget.Toast


class MainActivity : AppCompatActivity() {

    // Variabile di stato
    private var count = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // Collega il file Kotlin al file XML
        setContentView(R.layout.activity_main)

        val textView = findViewById<TextView>(R.id.text_counter)
        val btnIncrement = findViewById<Button>(R.id.button_increment)
        val btnReset = findViewById<Button>(R.id.button_reset) // Riferimento al nuovo ID

        btnIncrement.setOnClickListener {
            if (count < 10) {
                count++
                textView.text = count.toString()
            }

            // Controllo esplicito per disabilitare il componente
            if (count >= 10) {
                btnIncrement.isEnabled = false
                Toast.makeText(this, "Limite massimo raggiunto!", Toast.LENGTH_SHORT).show()
            }
        }
        btnReset.setOnClickListener {
            count = 0
            textView.text = "0"
            btnIncrement.isEnabled = true
        }
    }
}